### DISCUSSION TITLE
Connecting to SQL Server

### DISCUSSION THREAD

**Discussion by simulsys** [Budibase/budibase#2223] [Help] [answered]
Hi there,
I tried the configuration screen for SQL Server and it saved okay, but when I try to find how to attach it to a table, or form as with Postgresql or MySQL help screens my screen does not show the same commands and I get stuck. When I try to build and run a query it errors with a requests for a "Bucket" Key parameter? I have no idea what to do, so any help would be gratefully received, please?

Tally ho!
Andrew.

**Comment by mjashanks** [comment]
The postgres and MySql connectors are more advanced at the moment. They read tables from your database and provide lots of CRUD functionality, without having to write queries.

With MS SQL Server, you have to write all your own queries and provide any parameters. It would look a bit like this:

![image](https://user-images.githubusercontent.com/3524181/127887458-87e4a1c9-55d7-486c-acdd-ca420d98dead.png)

... you can see why we upgraded the other connectors! We will be upgrading MS SQL at some point, but I can't give you a date for this right now.

**Reply by simulsys** [reply]
Okay, thanks. A pity as SQL Server is our strategic database RDMS. Perhaps someday in the future.

**Reply by kevmodrome** [reply]
If you can't wait for it to get added you can contribute a new connector with support for this stuff. You can find the integrations here: https://github.com/Budibase/budibase/tree/master/packages/server/src/integrations

**Reply by simulsys** [reply]
I tried something similar but received this error condition notice:

![image](https://user-images.githubusercontent.com/6784401/128038997-79eae6be-ea0f-44b4-a621-4fd2114529a7.png)

**Reply by DanHaed** [reply]
I have the same problem. 
On Function READ i get `Error: Missing required key 'Bucket' in params    at KR.t `
on any other Function i get `Error: integration[queryVerb] is not a function    at KR.t`

**Reply by shogunpurple** [reply]
@DanHaed did you get this issue when you just tried to use the MySQL connector directly? Or did you also follow the guide above to build your own?

**Reply by DanHaed** [reply]
I tried to use the MSSQL connector directly.

**Reply by bloomt** [reply]
I am also getting Query Error: Missing Required key 'Bucket' in params error when attempting to run a query using the default MSSQL connector

**Comment by arpit314ad** [comment]
Hi...

I also get the same Error Msg.
Please suggest.

Arpit

**Comment by shogunpurple** [comment, marked as answer]
This issue has now been resolved and in the latest release. Thanks!
